### PR TITLE
Add screen sharing controls blog post + web platform documentation

### PR DESCRIPTION
**Pull request by beaufortfrancois**
FIX https://github.com/GoogleChrome/developer.chrome.com/issues/3672

Live preview: 
- https://deploy-preview-3869--developer-chrome-com.netlify.app/blog/avoiding-oversharing-when-screen-sharing/
- https://deploy-preview-3869--developer-chrome-com.netlify.app/docs/web-platform/screen-sharing-controls/
- https://deploy-preview-3869--developer-chrome-com.netlify.app/docs/web-platform/#shipped
- https://deploy-preview-3869--developer-chrome-com.netlify.app/authors/eladalon/

cc @eladalon1983

### CODE REVIEW

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+<figure>
+  {% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/SRzjjJ9AQVQVyjTCXskh.png", alt="Screenshot of the media picker prompt upon calling getDisplayMedia().", width="800", height="524" %}
+  <figcaption>
+    Chrome’s media picker prompt upon calling getDisplayMedia().
+  </figcaption>
+</figure>
+
+## Reordering the selection offered to the user {: #reordering }
+
+When [`getDisplayMedia()`] is called, users are presented with a media picker which allows them to share whichever display surface they would like any tab, window, or screen. Experience shows that the ordering of the offering affects the users’ selection. We are experimenting in Chrome&nbsp;107 on desktop with reordering the picker in order to nudge users towards tabs, which is the best option in the majority of cases.
```
Comment (eladalon1983) [inline site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:34]: This sentence becomes ungrammatical towards the end, with a single "like" trying to serve two purposes. (Any surface the users would like; options like...)

Comment (eladalon1983) [inline site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:34]: "On desktop" might be redundant, as `getDisplayMedia` is only supported on desktop.

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  <figcaption>
+    Chrome’s media picker (old vs. new).
+  </figcaption>
+</figure>
+
+The expected benefits are as follows:
+
+### Privacy
+
+Previously, screens were offered as the most prominent option. This is the least privacy-preserving option, as users may end up exposing more personal information than they had originally intended. Such as:
```
Comment (eladalon1983) [inline site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:47]: Maybe we should s/prominent/default

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## New screen sharing controls
+
+The new `displaySurface`, `surfaceSwitching`, `selfBrowserSurface`, and `systemAudio` options will allow you to tailor your screen sharing experience.
+
+Check out [Privacy-preserving screen sharing controls] for more information.
+
+## What's next {: #future }
+
+[Conditional Focus] will allow the capturing web app to instruct the browser to either switch focus to the captured display surface, or to avoid such a focus change.
```
Comment (eladalon1983) [inline site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:86]: Shall we mention suppressLocalAudioPlayback as well, do you think?

File: site/en/docs/web-platform/screen-sharing-controls/index.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+The values for the [`displaySurface`] option are:
+- `"browser"` for tabs.
+- `"window"` for windows.
+- `"monitor"` for screens.
+
+```js
+const stream = await navigator.mediaDevices.getDisplayMedia({
+  video: true,
+  displaySurface: "window", // Pre-select the "Window" pane in the media picker.
```
Comment (eladalon1983) [inline site/en/docs/web-platform/screen-sharing-controls/index.md:41]: Max line length exceeded on my device.
https://screenshot.googleplex.com/3PWu7qAeyqaff3W

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md (hunk 2 above)
Comment (beaufortfrancois) [reply site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:47]: Done

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md (hunk 1 above)
Comment (beaufortfrancois) [reply site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:34]: As agreed offline, I've removed ` any tab, window, or screen`

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md (hunk 3 above)
Comment (beaufortfrancois) [reply site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:86]: Sure.

File: site/en/docs/web-platform/screen-sharing-controls/index.md (hunk 4 above)
Comment (beaufortfrancois) [reply site/en/docs/web-platform/screen-sharing-controls/index.md:41]: Done

**Review by eladalon1983 — COMMENTED** [review]
[Here](https://screenshot.googleplex.com/ARWyT9PJZxR4iRH), would using a numbered list perhaps ensure that people:
1. ... would be less likely to miss the first bullet, which has no link?
2. ... would an easier time remembering their position in the doc when jumping back and forth with the other doc?

Whether you accept this suggestion or not, this is now LGTM. :-)

**Comment by beaufortfrancois** [comment]
> Whether you accept this suggestion or not, this is now LGTM. :-)

Fixed.

**Comment by rachelandrew** [comment]
@beaufortfrancois your glitch appears to be erroring in the docs.

**Comment by rachelandrew** [comment]
> > Whether you accept this suggestion or not, this is now LGTM. :-)
> 
> Fixed.

this should not be a numbered list, as per the style guide. https://developers.google.com/style/lists#types-of-lists please revert this to an unordered list as it is not a sequence of steps.

As a reminder, please make all changes in the Google doc and not on GitHub, and get any LGTMs there. We don't want to have to edit things twice.

**Comment by beaufortfrancois** [comment]
> @beaufortfrancois your glitch appears to be erroring in the docs.

@rachelandrew It "works on my machine". Which error do you see?

image

**Comment by rachelandrew** [comment]
It's working now, the first two times I loaded the page I got an error saying there was a problem starting the glitch and the glitch showed the Glitch homepage. It's likely better just to link to it as Glitch is incredibly unreliable, I've been moving a lot of demos to CodePen when it's possible to do so.

**Comment by beaufortfrancois** [comment]
> If CodePen works for the case then it's definitely a better option, it has rarely been unavailable. It isn't possible for every demo, but those where it is I'd suggest using it. We also have a team CodePen account so I can fork demos into that and then we are able to update them.

Good to know! Moving forward I'll try to use CodePen then for embedded content on d.c.c. and web.dev then. Thank you!

File: site/en/blog/avoiding-oversharing-when-screen-sharing/index.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+### Feature-completeness
+
+Tab sharing is more feature-complete:
+- Sharing tab audio is supported on all platforms; sharing system audio and window audio is only implemented on some platforms, at the time of writing.
+- When sharing tabs, complementary features enable closer collaboration with the captured content:
+  - [Capture Handle] allows the identification of the captured tab and establishment of a communications channel with it.
+  - [Region Capture] allows further scoping down of captured content.
+
+### Performance and User Experience
```
Comment (rachelandrew) [inline site/en/blog/avoiding-oversharing-when-screen-sharing/index.md:70]:
```suggestion
### Performance and user experience
```

**Review by rachelandrew — APPROVED** [review]
I just tweaked a heading, otherwise LGTM.

**Comment by guest271314** [comment]
> We are experimenting in Chrome&nbsp;107 on desktop with reordering the picker in order to nudge users towards tabs, which is the best option in the majority of cases.

Don't do that.

**Comment by guest271314** [comment]
I filed a specification issue and Chrome bug to stop doing that some time ago https://github.com/w3c/mediacapture-screen-share/issues/108. 

So it makes no sense to be capturing what the user picks.

**Comment by guest271314** [comment]
Note, the links at the bottom of the page are 404

> [displaySurface explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/displaysurface_constraint_explainer.md)
> [surfaceSwitching explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/surfaceswitching_explainer.md)
> [selfBrowserSurface explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/selfbrowsersurface_explainer.md)
> [systemAudio explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/systemaudio_explainer.md)

It would be helpful to also explain that if `echoCancellation` is _not_ passed as an audio constraint `channelCount` will be `1`.

**Comment by guest271314** [comment]
The `"systemAudio"` constraint is misleading: Chrome does not capture `speechSynthesis.speak(new SpeechSynthesisUtterance('test'))` called on the same Tab.

Additionally, I would explain that if `echoCancellation` audio constraint is not set `channelCount` of `MediaStreamTrack` of kind audio will have `channelCount` set to `1`.

**Comment by guest271314** [comment]
I proposed update to this here https://github.com/GoogleChrome/developer.chrome.com/pull/3947/commits/4b00c01208c381d40d6948d5ee74ba874b23f44c to explain to users `getDisplayMedia()` does not capture `speechSynthesis.speak()` executed on the same Tab, and unless `echoCancellation: false` constraint is passed `channelCount` of `MediaStreamTrack` of kinf audio will be `1`.

**Comment by beaufortfrancois** [comment]
> Note, the links at the bottom of the page are 404
> 
> > [displaySurface explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/displaysurface_constraint_explainer.md)
> > [surfaceSwitching explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/surfaceswitching_explainer.md)
> > [selfBrowserSurface explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/selfbrowsersurface_explainer.md)
> > [systemAudio explainer](https://github.com/eladalon1983/screen-share-explainers/blob/main/systemaudio_explainer.md)

Thank you very much @guest271314!
I've just sent https://github.com/GoogleChrome/developer.chrome.com/pull/3953 to address this issue

@eladalon1983 What do you think of the proposed note https://github.com/GoogleChrome/developer.chrome.com/commit/4b00c01208c381d40d6948d5ee74ba874b23f44c?

**Comment by eladalon1983** [comment]
This looks like a clarification specific to `getDisplayMedia()`, and out of place in articles or blog posts centered around new controls. I'd rather not add it. The key to clear communication is focus.

**Comment by guest271314** [comment]
Why withhold pertinent information?

**Comment by eladalon1983** [comment]
> Why withhold pertinent information?

Please refer back to the last sentence of my previous message.

**Comment by guest271314** [comment]
I am referring to that/ You naming convention is miisleading.

> and out of place in articles or blog posts centered around new controls.

The new controls are misleading. You cannot now claim you haven't been notified of that fact. `getDisplayMedia()` _does not_ capture system audio. At best it captures _tab_ audio, and only when the audio being played is not sourced from Web Speech API `speechSynthesis.speak(new SpeechSynthesisUtterance())`.

Some specification authors and users might think you can capture speech synthesis output and system audio with `getDisplayMedia()`, particular with the naming of the controls, which is not the case.

https://github.com/WebAudio/web-audio-api/issues/2478#issuecomment-1231734449
> `getDisplayMedia({audio: true})` allows capturing the device's output, as you note

The suggested note reduces the potential for confusion, thus satisfies the criteria of clear communication, so folks in the field won't be thinking system audio actually means that,

**Comment by eladalon1983** [comment]
@guest271314,

On the topic of API shape, if you refer to [this slide](https://docs.google.com/presentation/d/15iAIhzpaA6reKJBL-ecgYtic6ZKHEpKL5OK_sExTllc/edit#slide=id.g12268b98496_2_84), you'll see that I had proposed a different API shape, which I personally think would have been less confusing. However, working groups requires compromise, and `systemAudio` is the shape other members of the working group required. You're welcome to take this up with them.

On the topic of `speechSynthesis`, I see it as out-of-scope for this article. I don't think this is important for a significant enough chunk of this article's readership. That's my opinion.

**Comment by guest271314** [comment]
The `MediaStreamTrack` only having _1_ channel by default when `echoCancellation` was not set to `false` was not that trivial to figure out. I took a fair amount of testing to eleminate other reasons that is.

I am all for providing information as to the real state of affairs, not _withholding_ information, which leads to both users in the field and specification authors repeating the same mistakes and myths for years potentially.

Adding notes that `systemAudio` does not really mean enture system audio, rather only yab audio, and not `speechSynthesis.speak()` output; and that to get _2_ channels you need to explicitly use `echoCancellation: false` won't hurt anything.

New features are cool. As long as they are not named with misleading terms, and the author fails to disclose the limits of the feature.

**Comment by eladalon1983** [comment]
@guest271314, I have a family as well as a full time job. Kindly condense your communication into a small number of short messages, or else I am afraid I will not have the time to read them and respond.

**Comment by guest271314** [comment]
> On the topic of `speechSynthesis`, I see it as out-of-scope for this article. I don't think this is important for a significant enough chunk of this article's readership. That's my opinion.

Speech synthesis has gotten pushed to the side for many years now, in spite of multiple issues. 

You have failed to address the misleading naming convention. `getDisplayMedia()` **_does not_** capture system audio.

**Comment by guest271314** [comment]
> @guest271314, I have a family as well as a full time job. Kindly condense your communication into a small number of short messages, or else I am afraid I will not have the time to read them and respond.

So does everybody else.

You have no issue being verbose when it is your own opinion you are writing about.

**Comment by rachelandrew** [comment]
I'm going to lock this thread, it seems as if there are a mixture of issues here and some of them seem to be about the spec, this isn't the best venue for that discussion.

If there are specific errors with the content, then they can be raised as a new issue, please raise them as individual issues.

Spec issues should be raised against the spec.

**Pull request merged**